src/main/java/com/enonic/xp/app/welcome/WelcomePageScriptBean.java

```diff
@@ -420,22 +430,19 @@ private ApplicationJson toApplicationJson( final Application application )
         return builder.build();
     }
 
-    private List<TemplateApplicationJson> mustParseJsonFile( String filePath )
+    private List<TemplateApplicationJson> parseJsonFile( String filePath )
     {
         try (FileInputStream in = new FileInputStream( filePath ))
         {
             return objectMapper.readValue( in, new TypeReference<>()
             {
             } );
         }
-        catch ( JsonProcessingException e )
-        {
-            throw new UncheckedIOException( "Failed to parse template applications file at " + filePath, e );
-        }
         catch ( IOException e )
         {
-            throw new UncheckedIOException( "Failed to read template applications file at " + filePath, e );
+            LOG.warn( "Failed to parse template applications file at " + filePath, e );
```

Comment:
Does it mean that we now issue a warning in case template is missing?
It is better that to fail, but why not checking for file exists first? And not issue warning/fail if file does not even exist?

Reply:
There is a check if the template exists on [here](https://github.com/enonic/app-xp-welcome/blob/a2b09dc060d649e905548599b3cf6605b56d581d/src/main/java/com/enonic/xp/app/welcome/WelcomePageScriptBean.java#L272)
This function is called only if resource exists and then warns if json is malformed
But in our case template file is present and has `null` in it, so parsing fails.

Reply:
I do appreciate warning instead of throwing exception.

The issue is then in enonic-cli that creates a malformed template. Right?

Reply:
Correct

Reply:
https://github.com/enonic/cli-enonic/issues/564